core/trino-main/src/main/java/io/trino/execution/SqlStageExecution.java

```diff
@@ -504,24 +504,27 @@ private synchronized void updateTaskStatus(TaskStatus taskStatus)
             }
 
             TaskState taskState = taskStatus.getState();
-            if (taskState == TaskState.FAILED) {
-                RuntimeException failure = taskStatus.getFailures().stream()
-                        .findFirst()
-                        .map(this::rewriteTransportFailure)
-                        .map(ExecutionFailureInfo::toException)
-                        .orElse(new TrinoException(GENERIC_INTERNAL_ERROR, "A task failed for an unknown reason"));
-                stateMachine.transitionToFailed(failure);
-            }
-            else if (taskState == TaskState.ABORTED) {
-                // A task should only be in the aborted state if the STAGE is done (ABORTED or FAILED)
-                stateMachine.transitionToFailed(new TrinoException(GENERIC_INTERNAL_ERROR, "A task is in the ABORTED state but stage is " + stageState));
-            }
-            else if (taskState == TaskState.FLUSHING) {
-                flushingTasks.add(taskStatus.getTaskId());
-            }
-            else if (taskState == TaskState.FINISHED) {
-                finishedTasks.add(taskStatus.getTaskId());
-                flushingTasks.remove(taskStatus.getTaskId());
+
+            switch (taskState) {
+                case FAILED:
+                    RuntimeException failure = taskStatus.getFailures().stream()
+                            .findFirst()
+                            .map(this::rewriteTransportFailure)
+                            .map(ExecutionFailureInfo::toException)
+                            .orElse(new TrinoException(GENERIC_INTERNAL_ERROR, "A task failed for an unknown reason"));
+                    stateMachine.transitionToFailed(failure);
+                    break;
+                case ABORTED:
+                    // A task should only be in the aborted state if the STAGE is done (ABORTED or FAILED)
+                    stateMachine.transitionToFailed(new TrinoException(GENERIC_INTERNAL_ERROR, "A task is in the ABORTED state but stage is " + stageState));
+                    break;
+                case FLUSHING:
+                    flushingTasks.add(taskStatus.getTaskId());
+                    break;
+                case FINISHED:
+                    finishedTasks.add(taskStatus.getTaskId());
+                    flushingTasks.remove(taskStatus.getTaskId());
+                    break;
```

Comment:
please add `default`

Reply:
previously it was ignored, so it would be correct to still ignore it